application/common/models/User.php

```diff
@@ -40,6 +48,195 @@ public function rules()
     }
 
     /**
+     * Find or create local user. Fetch/update user data from personnel.
+     * @param string $username
+     * @return User
+     * @throws NotFoundHttpException
+     * @throws \Exception
+     * @throws NotFoundException
+     */
+    public static function findOrCreate($username)
+    {
+        /*
+         * If username looks like an email address, search by email
+         */
+        /** @var PersonnelUser $personnelUser */
+        if (substr_count($username, '@') > 0) {
+            $personnelUser = \Yii::$app->personnel->findByEmail($username);
+        } else {
+            $personnelUser = \Yii::$app->personnel->findByUsername($username);
+        }
+
+        $user = self::findOne(['employee_id' => $personnelUser->employeeId]);
+        if ( ! $user) {
+            $user = new User();
+            $user->employee_id = $personnelUser->employeeId;
```

Comment:
Is it possible for `$personnelUser` here to be `null` (such as if no such user was found in the personnel system)?